kernel/src/vm/memfd.rs

```diff
@@ -134,14 +225,32 @@ impl FileLike for MemfdFile {
     }
 
     fn seek(&self, pos: SeekFrom) -> Result<usize> {
-        do_seek_util(&self.inode, &self.offset, pos)
+        do_seek_util(&self.memfd_inode, &self.offset, pos)
     }
 
     fn fallocate(&self, mode: FallocMode, offset: usize, len: usize) -> Result<()> {
-        do_fallocate_util(&self.inode, self.status_flags(), mode, offset, len)
+        do_fallocate_util(&self.memfd_inode, self.status_flags(), mode, offset, len)
     }
 
     fn mappable(&self) -> Result<Mappable> {
-        Ok(Mappable::Inode(self.inode.clone()))
+        Ok(Mappable::Inode(self.memfd_inode.clone()))
     }
 }
+
+bitflags! {
+    pub struct MemfdFlags: u32 {
+        /// Close on exec.
+        const MFD_CLOEXEC = 1 << 0;
+        /// Allow sealing operations on this file.
+        const MFD_ALLOW_SEALING = 1 << 1;
+        /// Create in the hugetlbfs.
+        const MFD_HUGETLB = 1 << 2;
+        /// Not executable and sealed to prevent changing to executable.
+        const MFD_NOEXEC_SEAL = 1 << 3;
+        /// Executable.
+        const MFD_EXEC = 1 << 4;
+    }
+}
+
+// FIXME: This is a dummy implementation. Implement AnonInodeFs properly.
+static ANON_INODE_FS: Once<Arc<RamFs>> = Once::new();
```

Comment:
You can define this static variable in `fs()`, so you won't need to write the same FIXME comment twice.